testgrid/cmd/transfigure/cmd/transfigure.go

```diff
@@ -0,0 +1,362 @@
+	rootCmd.PersistentFlags().StringVar(&o.prowJobConfig, "prow_job_config", "", "Job Config path for your non-k8s Prow instance.")
+	rootCmd.PersistentFlags().StringVar(&o.testgridYaml, "testgrid_yaml", "", "TestGrid configuration directory or default file.")
+	rootCmd.PersistentFlags().StringVar(&o.repoSubdir, "repo_subdir", "", "The subdirectory in /config/testgrids/... to push to. Usually <github_org> or <github_org>/<github_repository>.")
+	for _, flag := range []string{"prow_config", "prow_job_config", "testgrid_yaml", "repo_subdir"} {
+		rootCmd.MarkPersistentFlagRequired(flag)
+	}
+
+	// Optional Flags
+	rootCmd.PersistentFlags().StringVar(&o.githubToken, "github_token", "", "A GitHub personal access token. A value of 'Test' will trigger a dry run.")
+	rootCmd.PersistentFlags().StringVar(&o.remoteForkRepo, "remote_fork_repo", "", "The name of the user's fork of kubernetes/test-infra, if it is not named test-infra.")
+	rootCmd.PersistentFlags().StringVar(&o.gitUser, "git_user", "", "The GitHub username to use. Pulled from github_token if not specified.")
+	rootCmd.PersistentFlags().StringVar(&o.gitEmail, "git_email", "", "The GitHub email to use. Pulled from github_token if not specified.")
+	rootCmd.PersistentFlags().BoolVar(&o.dryRun, "dry_run", false, "If true, no PR will be created.")
+}
+
+func createTempWorkingDir() error {
+	tempDir, err := ioutil.TempDir(".", "transfigure_")
+	if err != nil {
+		return wrapErrorOrNil("Error creating temp dir", err)
+	}
+	o.workingDir, err = filepath.Abs(tempDir)
+	if err != nil {
+		return wrapErrorOrNil("Error getting abs path for working dir", err)
+	}
+	log.Printf("Created temp directory %v/", tempDir)
+	return nil
+}
+
+func removeTempWorkingDir() {
+	if err := os.RemoveAll(o.workingDir); err != nil {
+		log.Printf("Error: Could not remove temp directory %v/", o.workingDir)
+	}
+	log.Printf("Removed temp directory %v/", o.workingDir)
+}
+
+func cloneK8sTestInfraRepo() error {
+	log.Print("Cloning kubernetes/test-infra into temp directory")
+	cmd := exec.Command("git", "clone", K8sTestInfraGitPath)
+	_, err := runCmd(cmd)
+	return wrapErrorOrNil("Error on git clone", err)
+}
+
+func createAndCheckoutGitBranch() error {
+	log.Print("Checking out branch " + Branch)
+	_, err := runCmd(exec.Command("git", "checkout", "-B", Branch))
+	return wrapErrorOrNil("Error on git checkout -B "+Branch, err)
+}
+
+func createRepoSubdirForTestgridYAML() {
+	o.testgridDir = filepath.Join(o.workingDir, "test-infra/config/testgrids", o.repoSubdir)
+	if _, err := os.Stat(o.testgridDir); os.IsNotExist(err) {
+		log.Printf("Directory %v does not exist; creating it.", o.testgridDir)
+		os.MkdirAll(o.testgridDir, 0755)
+	}
+}
+
+func generateTestgridYAML() error {
+	yamlPath, err := filepath.Abs(o.testgridYaml)
+	if err != nil {
+		return wrapErrorOrNil("Invalid testgrid yaml path "+o.testgridYaml, err)
+	}
+
+	cmd := exec.Command(
+		"/Users/joshuabone/go/src/k8s.io/test-infra/testgrid/cmd/transfigure/configurator",
```

Comment:
This can't be the correct path for all users, right? Does this work from an image?

Reply:
Done